src/host/VtInputThread.cpp

```diff
@@ -45,6 +46,27 @@ VtInputThread::VtInputThread(_In_ wil::unique_hfile hPipe,
 
     _pInputStateMachine = std::make_unique<StateMachine>(engine.release());
     THROW_IF_NULL_ALLOC(_pInputStateMachine.get());
+
+    _shutdownWatchdog = std::async(std::launch::async, [&] {
+        _shutdownEvent.wait();
+        if (_dwThreadId != 0)
+        {
+            wil::unique_handle threadHandle(OpenThread(STANDARD_RIGHTS_ALL | THREAD_TERMINATE, FALSE, _dwThreadId));
+            LOG_LAST_ERROR_IF_NULL(threadHandle.get());
+            if (threadHandle)
+            {
+                LOG_IF_WIN32_BOOL_FALSE(CancelSynchronousIo(threadHandle.get()));
+            }
+        }
+    });
+}
+
+VtInputThread::~VtInputThread()
+{
+    if (_shutdownEvent)
+    {
+        _shutdownEvent.SetEvent();
+    }

```

Comment:
Do we need to wait on the async so it doesn't get destructed with outstanding work? Is that an exception condition?

Reply:
I know if you call `std::thread::~thread` without detach/join it carps